table/server/under/hive/src/main/java/alluxio/table/under/hive/HiveLayout.java

```diff
@@ -100,14 +100,27 @@ public PartitionInfo getData() {
     return mPartitionStatsInfo;
   }
 
+  @Override
+  public String getLocation() {
+    return mPartitionInfo.getStorage().getLocation();
```

Comment:
instead of having a layout return a location, shouldn't `getTransformPlan` construct the jobs (and the appropriate locations), so external classes don't need to know about locations? Locations might be a concept specific to individual layouts.

Reply:
my two cents here.  I ran into a similar situation in updating table layout / location also because we have abstracted the location away from the layout.  IMO, location might mean different things to different layout but they all should have one? especially because `TransformContext` already exposes a location concept?

Reply:
currently, job services need to know the physical location. In the future, we can modify job service to use protobuf serialization for job configurations, in that way, the input and output of some jobs can be the protobuf definition of Layout, then we can completely get rid of `getLocation`.